internal/database/migration/stitch/git.go

```diff
@@ -8,14 +8,19 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strconv"
+	"strings"
 	"sync"
 
 	"github.com/sourcegraph/sourcegraph/internal/lazyregexp"
 	"github.com/sourcegraph/sourcegraph/internal/oobmigration"
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
-var gitTreePattern = lazyregexp.New("^tree .+:.+\n")
+var (
+	gitTreePattern = lazyregexp.New("^tree .+:.+\n")
+	versionPattern = lazyregexp.New(`v(\d+).(\d+)(?:.(\d+))`)
```

Comment:
This regex looks a little funny to me:
1) it looks like you want to escape the `.` so it matches a literal `.` rather than any character
2) you're using a non-capturing group for the trailing patch version, and it looks like you might have intended to make it an optional group instead.

```suggestion
	versionPattern = lazyregexp.New(`v(\d+)\.(\d+)(?:\.(\d+))?`)
```